### ISSUE TITLE
Installing “language-syntax-settings@0.2.1” failed.

### ISSUE THREAD

**Issue by exalted** [ardcore/atom-language-settings#2]
Atom `Version 0.73.0 (0.73.0)`:

```
> cson@1.4.5 preinstall /private/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/apm-install-dir-114217-24384-y2hxai/node_modules/language-syntax-settings/node_modules/cson
> node ./cyclic.js


> js2coffee@0.2.7 preinstall /private/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/apm-install-dir-114217-24384-y2hxai/node_modules/language-syntax-settings/node_modules/cson/node_modules/js2coffee
> npm run-script ./cyclic.js


> typechecker@2.0.8 preinstall /private/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/apm-install-dir-114217-24384-y2hxai/node_modules/language-syntax-settings/node_modules/cson/node_modules/extract-opts/node_modules/typechecker
> node ./cyclic.js


npm http GET https://registry.npmjs.org/jsesc
npm http GET https://registry.npmjs.org/cson
npm http GET https://registry.npmjs.org/underscore
npm http 304 https://registry.npmjs.org/underscore
npm http GET https://registry.npmjs.org/underscore/-/underscore-1.6.0.tgz
npm http 200 https://registry.npmjs.org/cson
npm http GET https://registry.npmjs.org/cson/-/cson-1.4.5.tgz
npm http 200 https://registry.npmjs.org/underscore/-/underscore-1.6.0.tgz
npm http 200 https://registry.npmjs.org/jsesc
npm http GET https://registry.npmjs.org/jsesc/-/jsesc-0.4.3.tgz
npm http 200 https://registry.npmjs.org/cson/-/cson-1.4.5.tgz
npm http 200 https://registry.npmjs.org/jsesc/-/jsesc-0.4.3.tgz
npm http GET https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/js2coffee
npm http GET https://registry.npmjs.org/requirefresh
npm http GET https://registry.npmjs.org/extract-opts
npm http 200 https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/coffee-script/-/coffee-script-1.6.3.tgz
npm http 200 https://registry.npmjs.org/extract-opts
npm http GET https://registry.npmjs.org/extract-opts/-/extract-opts-2.2.0.tgz
npm http 200 https://registry.npmjs.org/js2coffee
npm http GET https://registry.npmjs.org/js2coffee/-/js2coffee-0.2.7.tgz
npm http 200 https://registry.npmjs.org/requirefresh
npm http GET https://registry.npmjs.org/requirefresh/-/requirefresh-1.1.2.tgz
npm http 200 https://registry.npmjs.org/js2coffee/-/js2coffee-0.2.7.tgz
npm http 200 https://registry.npmjs.org/extract-opts/-/extract-opts-2.2.0.tgz
npm http 200 https://registry.npmjs.org/coffee-script/-/coffee-script-1.6.3.tgz
npm http 200 https://registry.npmjs.org/requirefresh/-/requirefresh-1.1.2.tgz
npm http GET https://registry.npmjs.org/typechecker
sh: npm: command not found
npm ERR! js2coffee@0.2.7 preinstall: `npm run-script ./cyclic.js`
npm ERR! Exit status 127
npm ERR! 
npm ERR! Failed at the js2coffee@0.2.7 preinstall script.
npm ERR! This is most likely a problem with the js2coffee package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run-script ./cyclic.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls js2coffee
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.1.0
npm ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--userconfig" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/.apmrc" "install" "/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/d-114217-24384-1uue21o/package.tgz" "--target=0.11.10" "--arch=x64"
npm ERR! cwd /private/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/apm-install-dir-114217-24384-y2hxai
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.4
npm ERR! code ELIFECYCLE
npm http 200 https://registry.npmjs.org/typechecker
npm http GET https://registry.npmjs.org/typechecker/-/typechecker-2.0.8.tgz
npm http 200 https://registry.npmjs.org/typechecker/-/typechecker-2.0.8.tgz
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /private/var/folders/s5/pbz1s7bn7jvgb96y5s4044wm0000gn/T/apm-install-dir-114217-24384-y2hxai/npm-debug.log
npm ERR! not ok code 0
```

**Comment by imchenxi** [comment]
Same issue here (10.9.2)

**Comment by kenkku** [comment]
This has something to do with atom's environment, I have the same problem when installing this package from the settings menu, but installing from command line works (`apm install language-syntax-settings`). Also, the problem does not seem to be specific to this package, but rather a problem with the atom js2coffee plugin (searls/atom-js2coffee#1).

**Comment by Blaisorblade** [comment]
Same issue here. I thought installing npm externally would help, but it didn't.

"Of course", ironically, npm install js2coffee (from outside) works just fine.

**Comment by imchenxi** [comment]
`apm install language-syntax-settings` in the Terminal works fine here!

**Comment by Blaisorblade** [comment]
Indeed, `apm install language-syntax-settings` works here too. Given the above error (`sh: npm: command not found`), my first guess would be to check the PATH environment variable inside Atom — inside OS X GUI applications it often has a different (and more "minimal") value (I learned this with Emacs). But I'm just guessing. Maybe this should be escalated also to Atom itself?

**Comment by ardcore** [comment]
I've published update including #4 provided kindly by @aki77. Package should work without problems now.

**Comment by Blaisorblade** [comment]
> I've published update including #4 provided kindly by @aki77. Package should work without problems now.

#4 seems unrelated, and indeed it doesn't help — I got right now the error below from trying to install 2.6.0.
@ardcore, are you unable to reproduce the problem? Do the analyses above make sense to you?

```
> cson@1.4.5 preinstall /private/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/apm-install-dir-114612-61759-1m8z0r5/node_modules/language-syntax-settings/node_modules/cson
> node ./cyclic.js


> js2coffee@0.2.7 preinstall /private/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/apm-install-dir-114612-61759-1m8z0r5/node_modules/language-syntax-settings/node_modules/cson/node_modules/js2coffee
> npm run-script ./cyclic.js


> typechecker@2.0.8 preinstall /private/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/apm-install-dir-114612-61759-1m8z0r5/node_modules/language-syntax-settings/node_modules/cson/node_modules/extract-opts/node_modules/typechecker
> node ./cyclic.js


npm http GET https://registry.npmjs.org/loophole
npm http GET https://registry.npmjs.org/cson
npm http GET https://registry.npmjs.org/underscore
npm http GET https://registry.npmjs.org/jsesc
npm http 200 https://registry.npmjs.org/loophole
npm http GET https://registry.npmjs.org/loophole/-/loophole-1.0.0.tgz
npm http 200 https://registry.npmjs.org/underscore
npm http 200 https://registry.npmjs.org/cson
npm http 200 https://registry.npmjs.org/loophole/-/loophole-1.0.0.tgz
npm http 200 https://registry.npmjs.org/jsesc
npm http GET https://registry.npmjs.org/extract-opts
npm http GET https://registry.npmjs.org/coffee-script
npm http GET https://registry.npmjs.org/requirefresh
npm http GET https://registry.npmjs.org/js2coffee
npm http 200 https://registry.npmjs.org/extract-opts
npm http 200 https://registry.npmjs.org/requirefresh
npm http 200 https://registry.npmjs.org/coffee-script
npm http 200 https://registry.npmjs.org/js2coffee
npm http GET https://registry.npmjs.org/typechecker
sh: npm: command not found
npm ERR! js2coffee@0.2.7 preinstall: `npm run-script ./cyclic.js`
npm ERR! Exit status 127
npm ERR! 
npm ERR! Failed at the js2coffee@0.2.7 preinstall script.
npm ERR! This is most likely a problem with the js2coffee package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     npm run-script ./cyclic.js
npm ERR! You can get their info via:
npm ERR!     npm owner ls js2coffee
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.3.0
npm ERR! command "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/bin/node" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/node_modules/npm/bin/npm-cli.js" "--globalconfig" "/Applications/Atom.app/Contents/Resources/app/apm/node_modules/atom-package-manager/.apmrc" "--userconfig" "/Users/pgiarrusso/.atom/.apmrc" "install" "/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/d-114612-61759-1vnpi96/package.tgz" "--target=0.11.10" "--arch=x64"
npm ERR! cwd /private/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/apm-install-dir-114612-61759-1m8z0r5
npm ERR! node -v v0.10.26
npm ERR! npm -v 1.4.4
npm ERR! code ELIFECYCLE
npm http 200 https://registry.npmjs.org/typechecker
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /private/var/folders/_7/hlxv4yv95x95vgnn416b4q4m0000gp/T/apm-install-dir-114612-61759-1m8z0r5/npm-debug.log
npm ERR! not ok code 0
```

**Comment by exalted** [comment]
I am closing this issue, but please feel free to re-open if interested. Thanks!
